installer/installer.go

```diff
@@ -56,8 +110,11 @@ func Install(art io.ReadCloser, dt string, key []byte, scrDir string,
 		log.Info("no public key was provided for authenticating the artifact")
 	}
 
-	if err := ar.RegisterHandler(rootfs); err != nil {
-		return errors.Wrap(err, "failed to register install handler")
+	// Important for the client to forbid artifacts types we don't know.
+	ar.ForbidUnknownHandlers = true
```

Comment:
No, the default is to accept unknown handlers, because that is what mender-artifact should do, which is where this comes from. Mender-artifact does not know anything about update modules, so in order to list them, it must accept them. The client however, needs to forbid this.

I could turn it into a property setter function instead, but it seemed overkill, and setting exported variables like this seems pretty common in Go (for example `exec.Cmd` struct).